modin/experimental/engines/omnisci_on_ray/io.py

```diff
@@ -288,3 +338,94 @@ def _prepare_pyarrow_usecols(cls, read_csv_kwargs):
                 raise NotImplementedError("unsupported `usecols` parameter")
 
         return usecols_md
+
+    @classmethod
+    def _read_csv_check_support(
+        cls,
+        read_csv_kwargs: ReadCsvKwargsType,
+    ) -> Tuple[bool, str]:
+        """
+        Check whatever or not passed parameters are supported by current modin.read_csv
+        implementation.
+
+        Parameters
+        ----------
+        read_csv_kwargs: ReadCsvKwargsType
+                Parameters of read_csv function.
+
+        Returns
+        -------
+        bool:
+            Whatever passed parameters are supported or not.
+        str:
+            Error message that should be raised if user explicitely set `engine="arrow"`.
+        """
+        filepath_or_buffer = read_csv_kwargs.get("filepath_or_buffer", None)
+        header = read_csv_kwargs.get("header", "infer")
+        names = read_csv_kwargs.get("names", None)
+        engine = read_csv_kwargs.get("engine", None)
+        skiprows = read_csv_kwargs.get("skiprows", None)
+        delimiter = read_csv_kwargs.get("delimiter", None)
+
+        if isinstance(filepath_or_buffer, str):
+            file_extension = os.path.splitext(filepath_or_buffer)[1]
+            if (
+                not cls.file_exists(filepath_or_buffer, only_locally=True)
+                or file_extension != ".csv"
+            ):
+                return False, cls.not_implemented_message
+        elif not cls.pathlib_or_pypath(filepath_or_buffer):
+            return False, cls.not_implemented_message
+
+        read_csv_signature = inspect.signature(cls.read_csv)
+        read_csv_default_kwargs = {
+            k: v.default
+            for k, v in read_csv_signature.parameters.items()
+            if v.default is not inspect.Parameter.empty
+        }
+        for arg in cls.unsupported_args:
+            if read_csv_kwargs[arg] != read_csv_default_kwargs[arg]:
+                return False, cls.not_implemented_message
```

Comment:
Can we at least specify which parameter causes the trouble here and in other places?

Reply:
Error messages were corrected.

Reply:
What if a user explicitly chooses Arrow engine and provides a wrong file name? Will he get NotImplementedError?